ngraph/core/include/openvino/core/node_output.hpp

```diff
@@ -33,13 +33,6 @@ class OPENVINO_API Output<Node> {
     /// \param index The index of the output.
     Output(Node* node, size_t index);
 
-    /// \brief Constructs a Output.
-    /// \param node A `shared_ptr` to the node for the output handle.
-    /// \param index The index of the output.
-    ///
-    /// TODO: Make a plan to deprecate this.
-    Output(const std::shared_ptr<Node>& node, size_t index);
```

Comment:
I would like to understand the story behind removing this constructor, I remember @GlebKazantaev https://github.com/openvinotoolkit/openvino/pull/7240#discussion_r696426571 and I'm not sure what is the answer. Also I can see that below there is still another constructor with `shared_ptr` param.

https://github.com/openvinotoolkit/openvino/blob/11591c550181ffcaef82fa91d722009c6234346f/ngraph/core/include/openvino/core/node_output.hpp#L36-L39